packages/core/src/shader/Shader.ts

```diff
@@ -89,31 +89,32 @@ export class Shader implements IReferable {
         throw "ShaderLab has not been set up yet.";
       }
 
-      const shaderInfo = Shader._shaderLab.parseShader(nameOrShaderSource);
-      if (shaderMap[shaderInfo.name]) {
-        console.error(`Shader named "${shaderInfo.name}" already exists.`);
+      const shaderStructInfo = Shader._shaderLab.parseShaderStruct(nameOrShaderSource);
+      if (shaderMap[shaderStructInfo.name]) {
+        console.error(`Shader named "${shaderStructInfo.name}" already exists.`);
         return;
       }
-      const subShaderList = shaderInfo.subShaders.map((subShaderInfo) => {
+      const subShaderList = shaderStructInfo.subShaders.map((subShaderInfo) => {
         const passList = subShaderInfo.passes.map((passInfo) => {
-          if (typeof passInfo === "string") {
+          if (passInfo.isUsePass) {
             // Use pass reference
-            const paths = passInfo.split("/");
+            const paths = passInfo.name.split("/");
             return Shader.find(paths[0])
               ?.subShaders.find((subShader) => subShader.name === paths[1])
               ?.passes.find((pass) => pass.name === paths[2]);
           }
 
           const shaderPass = new ShaderPass(
-            passInfo.name,
-            passInfo.vertexSource,
-            passInfo.fragmentSource,
+            passInfo.contents,
+            { vertexEntry: passInfo.vertexEntry, fragmentEntry: passInfo.fragmentEntry },
             passInfo.tags
           );
+          shaderPass.setName(passInfo.name);
```

Comment:
set name with `ShaderPass` constructor and delete `setName` API

Reply:
done